src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -812,28 +813,34 @@ FullQName AdvertiserMinMdns::GetCommissioningTxtEntries(const CommissionAdvertis
                                                                                            : &mQueryResponderAllocatorCommissioner;
 
     char txtVidPid[chip::Dnssd::kKeyVendorProductMaxLength + 4];
-    if (params.GetProductId().HasValue() && params.GetVendorId().HasValue())
+
+    const auto productId = params.GetProductId();
+    const auto vendorId  = params.GetVendorId();
```

Comment:
```suggestion
    const auto & productId = params.GetProductId();
    const auto & vendorId  = params.GetVendorId();
```

Reply:
All of these are like `Optional<uint16_t> GetVendorId() const { return mVendorId; }` today and the same (by value) after std::optional conversion.

I can make them return `const std::optional<uint16_t>&` however does that really save space? I am not sure they will ....